### DISCUSSION TITLE
Customizing filters on SOC

### DISCUSSION THREAD

**Discussion by ITgeevoSOC** [Security-Onion-Solutions/securityonion#6851] [Unsupported Versions] [answered]
Hi,

Is it possible to customize a filter in the alerts to have agent.ip?

If so, how is this done?

Thanks.

**Comment by dougburks** [comment, marked as answer]
I'm not sure I understand what you're asking but perhaps this snippet from https://docs.securityonion.net/en/2.3/alerts.html might help:
> When you click the arrow to expand a row in the Events table, it will show all of the individual fields from that event. Field names are shown on the left and field values on the right. When looking at the field names, there is an icon to the left that will add that field to the groupby section of your query. You can click on values on the right to bring up the context menu to refine your search or pivot to other pages.

**Reply by ITgeevoSOC** [reply]
Dear Doug,

To be more clear in the above question. It is possible to customize the default query bar to have as defaults Group By Name, Module which groups the alerts by rule.name and event.module, but to add also the Group: agent.ip module. if this is possible, how can we achieve that target?

Thanks in advance.

**Reply by dougburks** [reply]
You should be able to copy `/opt/so/saltstack/default/salt/soc/files/soc/alerts.queries.json` to `/opt/so/saltstack/local/salt/soc/files/soc/alerts.queries.json`, make your changes, and then restart SOC:
```
sudo so-soc-restart
```

**Reply by ITgeevoSOC** [reply]
Dear Doug,

Thanks. This has resolved our issue.
